File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    public int count = 0;
    
    public int solution(int[] numbers, int target) {
        backtracking(numbers, target, 0, 0);
        return count;
    }
    
    public void backtracking(int[] numbers, int target, int idx, int make) {
        if (numbers.length == idx) {
            if (make == target)
                count++;
            return;
        }
        
        backtracking(numbers, target, idx+1, make + numbers[idx]);
        backtracking(numbers, target, idx+1, make - numbers[idx]);
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
count = 0

def solution(numbers, target):
    backtracking(numbers, target, 0, 0)
    return count


def backtracking(numbers, target, idx, make):
    global count
    if idx == len(numbers):
        if target == make:
            count += 1
        return
    
    backtracking(numbers, target, idx+1, make+numbers[idx])
    backtracking(numbers, target, idx+1, make-numbers[idx])
    